exercises/practice/dnd-character/dnd_character_test.go

```diff
@@ -0,0 +1,70 @@
+package dndcharacter
+
+import "testing"
+
+func TestModifier(t *testing.T) {
+	for _, tc := range modifierTests {
+		t.Run(tc.description, func(t *testing.T) {
+			actual := Modifier(tc.input.Score)
+			if actual != tc.expected {
+				t.Fatalf("Modifier(%#v) = %#v, want %#v", tc.input, actual, tc.expected)
+			}
+		})
+	}
+}
+
+func TestAbility(t *testing.T) {
+	for i := 0; i < 10; i++ {
+		t.Run("should generate ability score within accepted range", func(t *testing.T) {
+			got := Ability()
+			assertAbilityScoreInRange(t, "random", got)
+		})
+	}
+}
+
+func TestGenerateCharacter(t *testing.T) {
+	t.Run("should generate a character with random ability scores", func(t *testing.T) {
+		character := GenerateCharacter()
+
+		assertAbilityScoreInRange(t, "Charisma", character.Charisma)
+		assertAbilityScoreInRange(t, "Strength", character.Strength)
+		assertAbilityScoreInRange(t, "Dexterity", character.Dexterity)
+		assertAbilityScoreInRange(t, "Wisdom", character.Wisdom)
+		assertAbilityScoreInRange(t, "Intelligence", character.Intelligence)
+		assertAbilityScoreInRange(t, "Constitution", character.Constitution)
+
+		if character.Hitpoints < 10 {
+			t.Fatalf("Character's base hitpoints are incorrect. Got %d", character.Hitpoints)
+		}
+	})
+}
+
+func BenchmarkAbility(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		Ability()
+	}
+}
+
+func BenchmarkCharacter(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		GenerateCharacter()
+	}
+}
+
+func BenchmarkModifier(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		Modifier(i)
+	}
+}
+
+func inAcceptedRange(score int) bool {
+	return score >= 3 && score <= 18
+}
+
+func assertAbilityScoreInRange(t testing.TB, ability string, score int) {
```

Comment:
```suggestion
func assertCharacterAbilityScoreInRange(t *testing.T, ability string, score int) {
```

Since we are no longer calling this function in `TestAbility`, we can make the name reflect that this function is specific to character abilities.

Using the interface `testing.TB` is unnecessary here, since we are only using it in the context of testing, so receiving the `*testing.T` is more intuitive to readers.